### ISSUE TITLE
Move HampelFilter from transformations to annotation module

### ISSUE THREAD

**Issue by RNKuhns** [alan-turing-institute/sktime#1331]
**Is your feature request related to a problem? Please describe.**
The `HampelFilter` was included in the transformation module prior to the addition of the annotation module. It is used to detect outliers, which is an annotation task. 

I've talked with @aiwalter who added it, and we both think it should be moved to the annotation module in the long-run. But we don't want to move it without being able to still use it (and other annotators) in a `ForecastingPipeline`.

**Describe the solution you'd like**
HampelFilter should be moved to the annotation module and refactored to comply with that interface. This should be done in conjunction with completion of #840, which will add a `ConditionalImputer` transformer that will allow annotators to be used
in a `ForecastingPipeline`. 

**Describe alternatives you've considered**
The alternative of leaving the `HampelFilter` in the transformation module is not ideal in the long-run because then we have estimators for the same task (detecting outliers) in more than one module.

**Additional context**
N/A

**Comment by RNKuhns** [comment]
@satya-pattnaik let me and @aiwalter know if there is anything we can help with in terms of #840. 

In terms of moving `HampelFilter` after #840 is complete, what is the expected sub-module structure for the `annotation` module? 

There are the related tasks of segmentation, outlier detection, anomaly detection, and so on. Do we want to group things that are used for certain tasks into a sub-module to make it easier for users? 

To me at least outlier detection seems like a specific case of anomaly/novelty detection (outliers being particular types of anomalies). What I don't really know about is what we mean when we say "segmentation". Is that different enough that the you'd be using one set of algorithms to solve that task and a different set for anomaly detection? 

If so, this would leave submodules organized be something like: `sktime.annotation.anomaly_detection` and `sktime.annotation.segmentation`. 

Interested to hear your thoughts so we can figure out where in the `annotation` module to move the `HampelFilter`.

**Comment by satya-pattnaik** [comment]
> if there is anything we can help with in terms of #840.

Yes I'll push some changes by coming Monday, and I might need help in adding the exogenous part as mentioned by @aiwalter  and to write up some unit tests.

**Comment by satya-pattnaik** [comment]
> If so, this would leave submodules organized be something like: `sktime.annotation.anomaly_detection` and `sktime.annotation.segmentation`.

Sounds good Outliers can come under Anomalies I guess so.

**Comment by aiwalter** [comment]
> Yes I'll push some changes by coming Monday, and I might need help in adding the exogenous part as mentioned by @aiwalter  and to write up some unit tests.

Sure lemme know, but what do you need to do according to exog?

**Comment by satya-pattnaik** [comment]
> but what do you need to do according to exog?

Please pardon me, I confused Z=DataFrame with exog. You can ignore that.
